lualib/rspamadm/ready.lua

```diff
@@ -0,0 +1,164 @@
+--[[
+Copyright (c) 2022, Vsevolod Stakhov <vsevolod@rspamd.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+]]--
+
+local argparse = require "argparse"
+local rspamd_http = require "rspamd_http"
+local rspamd_logger = require "rspamd_logger"
+local rspamd_upstream_list = require "rspamd_upstream_list"
+local lua_util = require "lua_util"
+
+local E = {}
+
+-- Define command line options
+local parser = argparse()
+    :name 'rspamadm ready'
+    :description 'Check if Rspamd controller is ready'
+    :help_description_margin(30)
+    :command_target('command')
+
+parser:option '-c --config'
+      :description 'Path to config file'
+      :argname('config_file')
+      :default(rspamd_paths['CONFDIR'] .. '/rspamd.conf')
+parser:option '-u --url'
+      :description 'URL of the Rspamd controller'
+      :argname('url')
+      :default('http://localhost:11334')
+parser:option '-t --timeout'
+      :description 'Total timeout in seconds'
+      :argname('timeout')
+      :default('60')
+parser:option '-i --interval'
+      :description 'Polling interval in seconds'
+      :argname('interval')
+      :default('1')
+parser:flag '--no-ssl-verify'
+      :description 'Disable SSL verification'
+      :argname('no_ssl_verify')
+parser:flag '-q --quiet'
+      :description 'Only output errors'
+      :argname('quiet')
+parser:flag '-v --verbose'
+      :description 'Output more information'
+      :argname('verbose')
+
+local http_params = {
+  config = rspamd_config,
+  ev_base = rspamadm_ev_base,
+  session = rspamadm_session,
+  resolver = rspamadm_dns_resolver,
+}
+
+local function load_config(config_file)
```

Comment:
We don't need to load config I suppose. It could be used for finding controller address but I think having it reconfigurable & giving it a sensible default is fine.